crypto/fipsmodule/bn/internal.h

```diff
@@ -790,6 +790,177 @@ void bn_little_endian_to_words(BN_ULONG *out, size_t out_len, const uint8_t *in,
 // leading zeros.
 void bn_words_to_little_endian(uint8_t *out, size_t out_len, const BN_ULONG *in, const size_t in_len);
 
+
+// Naming convention for the following functions:
+//
+//   * amm: Almost Montgomery Multiplication
+//   * ams: Almost Montgomery Squaring
+//   * 52xZZ: data represented as array of ZZ digits in 52-bit radix
+//   * _x1_/_x2_:  1 or 2 independent inputs/outputs
+//   * ifma256: uses 256-bit wide IFMA ISA (AVX512_IFMA256)
+//
+//
+// Almost Montgomery Multiplication (AMM) for 20-digit number in radix
+// 2^52.
+//
+// AMM is defined as presented in the paper [1].
+//
+// The input and output are presented in 2^52 radix domain, i.e.
+// |res|, |a|, |b|, |m| are arrays of 20 64-bit qwords with 12 high
+// bits zeroed.  |k0| is a Montgomery coefficient, which is here k0 =
+// -1/m mod 2^64
+//
+// NB: the AMM implementation does not perform "conditional"
+// subtraction step specified in the original algorithm as according
+// to the Lemma 1 from the paper [2], the result will be always < 2*m
+// and can be used as a direct input to the next AMM iteration.  This
+// post-condition is true, provided the correct parameter |s| (notion
+// of the Lemma 1 from [2]) is chosen, i.e.  s >= n + 2 * k, which
+// matches our case: 1040 > 1024 + 2 * 1.
+//
+// [1] Gueron, S. Efficient software implementations of modular
+//     exponentiation.  DOI: 10.1007/s13389-012-0031-5
+// [2] Gueron, S. Enhanced Montgomery Multiplication.  DOI:
+//     10.1007/3-540-36400-5_5
+void ossl_rsaz_amm52x20_x1_ifma256(BN_ULONG *res, const BN_ULONG *a,
```

Comment:
can we get rid of the `ossl` prefix in all these functions? I don't mind explicitly stating that this code was copied from OpenSSL, it's just the prefix doesn't make sense in our case.